pkg/minikube/machine/client.go

```diff
@@ -18,23 +18,150 @@ package machine
 
 import (
 	"crypto/tls"
-	"os"
+	"encoding/json"
+	"fmt"
+	"net"
+	"path/filepath"
 	"time"
 
-	"net"
+	"k8s.io/minikube/pkg/minikube/constants"
 
-	"github.com/docker/machine/drivers/hyperv"
 	"github.com/docker/machine/drivers/virtualbox"
-	"github.com/docker/machine/drivers/vmwarefusion"
+	"github.com/docker/machine/libmachine"
 	"github.com/docker/machine/libmachine/auth"
 	"github.com/docker/machine/libmachine/cert"
-	"github.com/docker/machine/libmachine/check"
-	"github.com/docker/machine/libmachine/drivers/plugin"
-	"github.com/docker/machine/libmachine/drivers/plugin/localbinary"
+	"github.com/docker/machine/libmachine/drivers"
+	"github.com/docker/machine/libmachine/drivers/rpc"
+	"github.com/docker/machine/libmachine/engine"
 	"github.com/docker/machine/libmachine/host"
-	"github.com/golang/glog"
+	"github.com/docker/machine/libmachine/persist"
+	"github.com/docker/machine/libmachine/swarm"
+	"github.com/docker/machine/libmachine/version"
+	"github.com/pkg/errors"
+)
+
+type driverGetter func(string, []byte) (drivers.Driver, error)
+
+type ClientType int
+
+const (
+	ClientTypeLocal ClientType = iota
+	ClientTypeRPC
 )
 
+// Gets a new client depending on the clientType specified
+// defaults to the libmachine client
+func NewAPIClient(clientType ClientType) libmachine.API {
+	storePath := constants.Minipath
+	certsDir := constants.MakeMiniPath("certs")
+
+	switch clientType {
```

Comment:
Yeah forgot about this switch statement.  Will do the same as I did for getDrivers